### ISSUE TITLE
use process.hrtime()

### ISSUE THREAD

**Issue by ryanstevens** [broofa/node-uuid#109]
I'm wondering if reliability and / or determinism would improve if high resolution timers were used instead of JavaScript Date object, which has been known to be fairly inaccurate and flat out wrong http://stackoverflow.com/questions/6233927/microsecond-timing-in-javascript

Thinking about putting this line as a fallback
https://github.com/broofa/node-uuid/blob/master/uuid.js#L130

and doing something like this (pseudo code- formats for these are all different) 
`process.hrtime() || window.performance.now() || (new Date().getTime() * 1000)`

**Comment by broofa** [comment]
The 3 issues we care about with uuid clock ...

*Are UUIDs unique?*  the `clock_seq` property guarantees uniqueness regardless of clock resolution.  `hrtime()` does not improve on this.

*What's the timestamp resolution?*  UUID timestamps support 100-nsec resolution, but Date.now() only supports msec resolution (and in some cases is even less accurate in practice. e.g. IIRC, legacy versions of IE only had ~15-20 msecs resolution).   I believe that's less an issue these days, and in node.js environments we can assume 1 msec resolution.  `hrtime()` would improve this, "but..."

*Is the time accurate?* The docs for `hrtime` state:

> It is relative to an arbitrary time in the past

I.e. `hrtime()` is only useful as an offset relative to a Date.time() snapshot, meaning it's no more accurate than Date.time().  And Date.time() is generally only as precise as the [NTP protocol](http://en.wikipedia.org/wiki/Network_Time_Protocol) which has a practical error of ~10-100ms, depending on network structure.

So basically the one advantage of process.hrtime() is that it has sub-msec resolution, but this is in a context where absolute accuracy is, at best, ~10 msecs... and that's in the best case.  Basically this isn't something I'm inclined to worry about until we have real-world cases where it matters.

**Comment by broofa** [comment]
Also, "Hi!" :)

**Comment by treemore** [comment]
https://jsperf.com/date-now-vs-new-date
i think can use Date.now()  instead of new Date().getTime()
